controllers/metal3.io/host_state_machine.go

```diff
@@ -317,6 +326,8 @@ func (hsm *hostStateMachine) handleProvisioning(info *reconcileInfo) actionResul
 	actResult := hsm.Reconciler.actionProvisioning(hsm.Provisioner, info)
 	if _, complete := actResult.(actionComplete); complete {
 		hsm.NextState = metal3v1alpha1.StateProvisioned
+	} else if result, failed := actResult.(actionFailed); failed && result.ErrorType == metal3v1alpha1.PreparingError {
+		hsm.NextState = metal3v1alpha1.StatePreparing
```

Comment:
See below the comments https://github.com/metal3-io/baremetal-operator/pull/292/files#r553262296